etl/etl.py

```diff
@@ -3,51 +3,105 @@
+
+
+def path_to_name(x: list, output_names: List[str]):
+    """Takes in a list of full addresses, extracts the package name from each of the address, and returns a list of package names."""
+    names = []
+    for p in x:
+        if p != None:
+            names.append(
+                "-".join(p.split("-")[1:-1])
+                if p.split("-")[-1] in output_names
+                else p[44:]
+            )
+    return names
+
+
+def path_to_outputpath(dataframe: pd.DataFrame, path: str, name: str) -> Optional[str]:
+    """Analyzes the full address of a package, and maps it to the output path with the same package.
+
+    Args:
+        dataframe: The dataframe consisting of outputPath, outputPathAll, and other attributes.
+        path: The full address of a package.
+        name: The package name of which the full address 'path' refers to.
+    """
+    df_name = dataframe.query("name == @name")
+    mask = df_name.outputPathAll.apply(lambda x: path in [ele["path"] for ele in x])
+    if df_name[mask].empty:
+        return None
+    return df_name[mask].outputPath.iloc[0]
+
+
 def unique_insert_node(g, row):
-    g.V().has("package", "outputPath", str(row["outputPath"])) \
-        .fold() \
-        .coalesce(__.unfold(), \
-                  __.addV("package") \
-                  .property("outputPath", str(row["outputPath"])) \
-                  .property("path", row["path"]) \
-                  .property("pname", row["pname"]) \
-                  .property("version", row["version"])) \
-        .iterate()
-
-def unique_insert_edge(g, row, target):
+    g.V().has("package", "outputPath", str(row["outputPath"])).fold().coalesce(
+        __.unfold(),
+        __.addV("package")
+        .property("outputPath", str(row["outputPath"]))
+        .property("path", row["path"])
+        .property("pname", row["pname"])
+        .property("name", row["name"])
+        .property("version", row["version"])
+        .property("brokenState", row["brokenState"])
+        .property("license", row["license"]),
+    ).iterate()
+
+
+def unique_insert_edge(g, row, target, label):
     if g.V().has("package", "outputPath", target).toList() != []:
-        g.V().has("package", "outputPath", row["outputPath"]) \
-             .as_("v") \
-             .V().has("package", "outputPath", target) \
-                 .coalesce(__.inE("dependsOn") \
-                           .where(__.outV().as_("v")), \
-                           __.addE("dependsOn").from_("v") \
-                           .property("label", "buildInputs")) \
-                 .iterate()
+        g.V().has("package", "outputPath", row["outputPath"]).as_("v").V().has(
+            "package", "outputPath", target
+        ).coalesce(
+            __.inE("dependsOn").where(__.outV().as_("v")),
+            __.addE("dependsOn").from_("v").property("label", label),
+        ).iterate()
 
 
 def gremlin_queries(dataframe):
```

Comment:
Could we rename this function to more precicesly represent what it is doing (e.g. `ingest_graph` or `write_to_gremlin` or something similar)? It would also be nice if we could add a docstring to this method to describe the DataFrame we expect as the input.